skimage/filters/_gabor.py

```diff
@@ -13,11 +15,187 @@ def _sigma_prefactor(bandwidth):
+
+    Notes
+    -----
+    Components ``0``, ``1``, ``...``, ``n`` of the quasipolar coordinates
+    correspond to dimensions ``0``, ``1``, ``...``, ``n`` or ``M``, ``...``,
+    ``N``, ``P``.
+
+    For a standard ``xy``-coordinate plane, components ``1`` and ``0``
+    correspond to ``x`` and ``y``, respectively.
+
+    For a standard ``xyz``-coordinate plane, components ``1``, ``0``, and ``2``
+    correspond to ``x``, ``y``, and ``z``, respectively.
+    """
+    axes = np.size(thetas) + 1
+    coords = r * np.ones(axes)
+
+    for which_theta, theta in enumerate(thetas[::-1]):
+        sine = np.sin(theta)
+        theta_index = axes - which_theta - 1
+
+        for axis in range(theta_index):
+            coords[axis] *= sine
+
+        coords[theta_index] *= np.cos(theta)
+
+    return coords
+
+
+def _gaussian(image, center=0, sigma=1, ndim=2):
+    """Generates gaussian envelope.
+
+    Parameters
+    ----------
+    image : non-complex array
+        Image to seed the filter with.
+    center : scalar or vector, optional
+        Coordinates to center the image with. Defaults to 0.
+    sigma : scalar or vector, optional
+        Spatial dimensions of the envelope. Defaults to 1.
+    ndim : int, optional
+        Dimensionality of the envelope to create. Defaults to 2.
+
+    Returns
+    -------
+    gauss : (``ndim``, ``ndim``)
+
+    References
+    ----------
+    .. [1] Do CB. 2008. The Multivariate Gaussian Distribution. Stanford
+           University (CS 229): Stanford, CA.
+           http://cs229.stanford.edu/section/gaussians.pdf
+    """
+    sigma_prod = np.prod(sigma)
+
+    # normalization factor
+    norm = (2 * np.pi) ** (ndim / 2)
+    norm *= sigma_prod
+
+    # center image
+    image = image - center
+
+    # gaussian envelope
+    gauss = np.exp(-0.5 * np.sum(image ** 2) / sigma_prod ** 2)
+
+    return gauss / norm
+
+
+def _rotation(src_axis, dst_axis):
+    """Generates an matrix for the nD rotation of one axis to face another.
+
+    Parameters
+    ----------
+    src_axis : (N,) matrix
+        Vector representation of the axis that will be rotated.
+
+    dst_axis : (N,) matrix
+        Vector representation of the axis to rotate to.
+
+    Returns
+    -------
+    M : (N, N) array
+        Matrix that rotates ``src_axis`` to coincide with ``dst_axis``.
+
+    References
+    ----------
+    .. [1] Ognyan Ivanov Zhelezov. One Modification which Increases Performance
+           of N-Dimensional Rotation Matrix Generation Algorithm. International
+           Journal of Chemistry, Mathematics, and Physics, Vol. 2 No. 2, 2018:
+           pp. 13-18. https://dx.doi.org/10.22161/ijcmp.2.2.1
+
+    Examples
+    --------
+    >>> X = np.asarray([1, 0])
+    >>> Y = np.asarray([0.5, 0.5])
+
+    >>> M = _rotation(X, Y)
+    >>> Z = np.matmul(M, Y[..., np.newaxis])
+
+    >>> np.allclose(Z, Y[np.newaxis])
+    True
+    """
+    X = np.array(src_axis)
+    Y = np.array(dst_axis)
+
+
+    def rot(axis, diffs):
```

Comment:
I don't see a reason why this function definition should be nested. Take it out, make it a private function, and provide a docstring about what it does exactly. =)